src/Aws/AwsStorageProvider.php

```diff
@@ -167,4 +174,29 @@ public function executeCopyRequests($requests, $callback)
             ->promise()
             ->wait();
     }
+
+    /**
+     * Get a handler stack containing the retry middleware configuration.
+     *
+     * @return \GuzzleHttp\HandlerStack
+     */
+    private function retryHandler()
+    {
+        $stack = HandlerStack::create();
+        $stack->push(Middleware::retry(function (int $retries, RequestInterface $request, ResponseInterface $response = null) {
```

Comment:
This will retry a "single" asset correct? So, in a batch of 10, we can have up to 3 * 10 requests correct?